luigi/db_task_history.py

```diff
@@ -79,13 +78,32 @@ def _session(self, session=None):
     def __init__(self):
         config = configuration.get_config()
         connection_string = config.get('task_history', 'db_connection')
-        self.engine = sqlalchemy.create_engine(connection_string)
+        connection_options = self._prepare_connection_options(config)
+
+        self.engine = sqlalchemy.create_engine(connection_string, **connection_options)
         self.session_factory = sqlalchemy.orm.sessionmaker(bind=self.engine, expire_on_commit=False)
         Base.metadata.create_all(self.engine)
         self.tasks = {}  # task_id -> TaskRecord
 
         _upgrade_schema(self.engine)
 
+    CONNECTION_OPTIONS_MAPPING = {
+            'db_pool_size': {'type': 'int', 'name': 'pool_size'},
+            'db_max_overflow': {'type': 'int', 'name': 'max_overflow'},
+            'db_pool_timeout': {'type': 'int', 'name': 'pool_timeout'},
+            'db_pool_recycle': {'type': 'int', 'name': 'pool_recycle'},
+            }
+
+    def _prepare_connection_options(self, config):
+        options = {}
+        for (config_key, column_settings) in six.iteritems(self.CONNECTION_OPTIONS_MAPPING):
+            column_name = column_settings['name']
+            if column_settings['type'] == 'int':
```

Comment:
This is a `_private` function, yet you anticipate something else than `'int'`, but `'int'` is hard-coded in the code already. This can be simplified.